src/openms/source/TRANSFORMATIONS/FEATUREFINDER/FeatureFinderIdentificationAlgorithm.cpp

```diff
@@ -504,8 +504,16 @@ namespace OpenMS
     Int charge = 0;
     if (extract_charge)
     {
-      Size pos_hash = target_id.find('#', pos_slash + 2);
-      charge = target_id.substr(pos_slash + 1, pos_hash).toInt();
+      if (target_id.find("#") != std::string::npos) // "PEP:XXXXX/3#1
+      {
+        Size pos_hash = target_id.find('#', pos_slash + 2);
+        size_t charge_size = pos_hash - pos_slash - 1;
+        charge = target_id.substr(pos_slash + 1, charge_size).toInt();
+      }
+      else // "PEP:XXXXX/2
+      {
+        charge = target_id.substr(pos_slash + 1, target_id.size()).toInt();
+      }
```

Comment:
not very efficient as it parses the string multile times

Reply:
Tried to clean it up.